cirq-core/cirq/protocols/apply_unitary_protocol.py

```diff
@@ -274,6 +280,10 @@ def _apply_unitary_(
         """
 
 
+def _strat_apply_unitary_from_unitary_(val: Any, args: ApplyUnitaryArgs) -> Optional[np.ndarray]:
```

Comment:
Maybe do it the other way so you don't change the interface of `_strat_apply_unitary_from_unitary` and introduce a new `_apply_unitary_from_matrix` which only accepts a `matrix: np.ndarray` and `args: ApplyUnitaryArgs ` and multiplies the `matrix` into `args`. Then both  `_strat_apply_unitary_from_unitary` and `_strat_apply_unitary_from_unitary` will call `_apply_unitary_from_matrix`. This way, you don't need to add the unnecessary check in `_strat_apply_unitary_from_unitary` that the provided `matrix` is None or not.

Reply:
done